### ISSUE TITLE
Map Cellular: Map without Location Services

### ISSUE THREAD

**Issue by Timorttr** [christianrowlands/android-network-survey#65]
Would it be possible for the Cellular map to jump to the current CellId when the localization service is deactivated?
GNSS is deactivated, but jump to the cell tower on the map anyway.
Cell tower in the center of the map.

**Comment by christianrowlands** [comment]
That is a good idea. I had not thought of that. I will see how much effort is involved to make that change.

**Comment by christianrowlands** [comment]
I might be misunderstanding this one a bit @Timorttr . I took a stab at implementing this, but as I was doing it I realized that I can't get the current serving cell without having location services enabled. The reason Android takes this approach is that knowing the serving cell means an app can figure out your rough location, which is a privacy concern. Therefore, to get the current serving cell, location services have to be enabled.

In short, if location services are disabled by the user then I can't get any information, including the serving cell, which prevents me from being able to center the map on the cell tower.

Am I missing something with another use case?

**Comment by christianrowlands** [comment]
One thing I could do is store the last viewed location, so that way the next time the map is opened, I could zoom to that location. Is that of interest to you?

**Comment by Timorttr** [comment]
The last seen function, or change to the current cell (if the cell phone cell exists) in the map would be nice.
It is interesting for me when I have not yet found a position, or indoors in a building.

**Comment by Timorttr** [comment]
If I am in a building and have no GPS, go to the position of the cell id and its range on the map.

**Comment by Timorttr** [comment]
Perhaps another topic.
Would it be possible to use beaconDB with wifi or cellid for localization? As an option.

**Comment by christianrowlands** [comment]
@Timorttr , has the change to restore the last saved map view location been good enough? I don't really want to go the route of implementing my own location query if I can avoid it. Added complexity typically results in more edge case bugs that I have to hunt down.

**Comment by christianrowlands** [comment]
@Timorttr . Since I am restoring the last map view location when opening the map, I am going to close this issue. Feel free to comment if that is not working well enough for you, but I am hoping that solves the problem enough to make it work
